flaml/model.py

```diff
@@ -781,4 +793,13 @@ def _preprocess(self, X):
                 raise ValueError(
                     "kneighbor requires at least one numeric feature")
             X = X.drop(cat_columns, axis=1)
+        elif X.dtype.kind not in 'buif':

```

Comment:
in this case, we do not need to check whether X is np array? Can input X be any other type except Dataframe or numpy array?

Reply:
Thanks. Fixed.